feat: ej/login flow

lightlii opened the pull request:
Primarily view components around the authentication screens.

I'm not sure how but I seem to have messed something up in the rebase because we have all the commit from your 'set up typescript' branch @ErikSin. Also I'm not entirely sure why but I'm getting ts errors from the theme system:
`Property 'primary' does not exist on type 'Theme'`
That wasn't happening when I was working on it a few weeks back as far as I remember, but started since rebasing against `develop`. Can look at that together / discuss strategy for this kind of thing tomorrow when we call (Erik)

ErikSin committed:
chore: set up typescript

ErikSin committed:
chore: set up typescript

ErikSin committed:
feat: created readme

ErikSin committed:
chore: removed readme

ErikSin committed:
chore: added to dev depdencies

ErikSin committed:
chore: added to dev depdencies

Commit:
chore: added basline theme

Commit:
feat: add split-layout component

Commit:
chore: add firebaseConfig type

Commit:
chore: add back in router

Commit:
misc: file missed from last commit

Commit:
chore: adding in reach-router types

Commit:
fix: fix typing for react-intl provider

Commit:
feat: add AuthPanel component (generic layout for auth)

Commit:
chore: styles for AuthPanel

Commit:
feat: add main login screen

Commit:
chore: add resuable input components

ErikSin committed:
chore: updated firestore and firestore hook

lightlii committed:
fix: convert colour names in theme to CONSTANT_CASE

lightlii committed:
chore: extracted firebase initialisation to own file to export firebaseApp

lightlii committed:
fix: fix breaking changes from bad conflict resolution

ErikSin commented on src/components/App/types.ts:
```diff
@@ -0,0 +1,6 @@
+export type AuthorizedProps = {
```
Delete file

ErikSin commented on src/components/AuthPanel/index.tsx:
```diff
@@ -0,0 +1,48 @@
+import { Container, Link, Typography } from "@mui/material"
```
Use translations instead of hardcoded strings

ErikSin commented on src/theme.ts:
```diff
@@ -1,35 +1,37 @@
 import { createTheme } from "@mui/system";
```
According to [MUI theme docs](https://mui.com/material-ui/customization/theming/), you need to use typescripts [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation) when you are defining new things in the theme. In this case, out of the box there is a typing for `pallete.primary.main` but you need to override it with module augmentation since you are not using the `main` attribute.

eg:
``` ts
declare module '@mui/material/styles'{
  interface Theme{
    palette:{primary:string}
  }
}
```

ErikSin commented on src/components/TextInput/index.tsx:
```diff
@@ -0,0 +1,26 @@
+  const theme = useTheme()
+
+  const inputSx = {
+    backgroundColor: theme.white,
```
I don't believe `white` is defined in your `theme` object

ErikSin commented on package.json:
```diff
@@ -40,8 +41,8 @@
     "@types/reach__router": "^1.3.11",
-    "@types/react": "^17.0.52",
-    "@types/react-dom": "^17.0.18",
+    "@types/react": "^18.0.25",
+    "@types/react-dom": "^18.0.9",
```
I don't know why this didn't end up in the original pr we did, but it should be types 18

ErikSin commented on src/components/AuthPanel/index.tsx:
```diff
@@ -0,0 +1,48 @@
+  )
+}
+
+export default AuthPanel
```
convert to named

ErikSin commented on src/components/FormButton/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import EastIcon from '@mui/icons-material/East'
+import { Button, ButtonTypeMap } from '@mui/material'
```
get rid of unused import `ButtonTypeMap`

ErikSin commented on src/components/FormButton/index.tsx:
```diff
@@ -0,0 +1,26 @@
+  </Button>
+)
+
+export default FormButton
```
convert to named

ErikSin commented on src/components/TextInput/index.tsx:
```diff
@@ -0,0 +1,26 @@
+import { BaseTextFieldProps, FormLabel, Stack, useTheme, TextFieldProps, TextField } from "@mui/material"
```
Get rid of unused imports `BaseTextFieldProps` and `TextField`

ErikSin commented on src/components/TextInput/index.tsx:
```diff
@@ -0,0 +1,26 @@
+  )
+}
+
+export default TextInput
```
convert to named

ErikSin commented on src/firebase-init.ts:
```diff
@@ -0,0 +1,18 @@
+import { initializeApp } from "firebase/app";
```
You can just delete this file

ErikSin commented on src/types/index.ts:
```diff
@@ -0,0 +1,12 @@
+export type firebaseConfigType = {
```
delete file

ErikSin commented on src/views/Login/SignInForm.tsx:
```diff
@@ -0,0 +1,121 @@
+  Stack,
+  Checkbox,
+  FormControlLabel,
+  Typography,
+  Link,
+  FormLabel,
```
remove unused import `typography` and `form label`

ErikSin commented on src/views/Login/SignInForm.tsx:
```diff
@@ -0,0 +1,121 @@
+        name="email"
+        autoComplete="email"
+        autoFocus
+        // error={!!isEmailError}
+        // helperText={isEmailError && formatMessage(msgs[error.code])}
```
is this commented out for a reason?

ErikSin reviewed (commented):
I think that it was originally re-based off the `scaffolding` branch. Then we made significant changes to `scaffolding`, and merged `scaffolding` into develop. Then we re-based this off develop and since develop is a completely different branch, it does not see the rebased work off` scaffolding` and thinks all the old edits are new work.

In order to do stacked PRs efficiently, we need to be merging into develop more often, before we get too ahead. I think what is going to happen is that these rebase problems are going to get worse since we are making significant changes, but still working on a couple PRs ahead. I suggest that we focus on getting this PR ready, then we rebase your next pr, and merge this one. And then we only try to stay 1 (maybe 2) prs ahead.

Secondly, can you run prettier. that should get rid of some of these file changes. The automation only works on staged files, so we need to run that command until we get to a point that your PRs have the automation re-based into it **before** the file is staged.

I also added comments about the themeing/typescript errors. Try fixing those and see if it is running. Currently I am just getting a blank screen when I run it

lightlii commented:
First of all, sorry this PR is a complete mess. More on that after.

> I think that it was originally re-based off the `scaffolding` branch.

Yeah that makes sense, that explains why the commit tree is looking so weird. 

> In order to do stacked PRs efficiently, we need to be merging into develop more often, before we get too ahead.

Yes agreed, will work on that.

> I also added comments about the themeing/typescript errors.

Yeah we should talk about this later today but one of the issues I have at the moment is that a lot of the comments you've left here, are things that are fixed in later branches eg. I added the typescript module augmentation stuff [here](https://github.com/digidem/mapeo-webmaps/commit/e4430b10c33a37aec4bc9f9481c208b7468708bb). I'm not really sure how I've ended up with with references to 'theme.white' etc in this branch though because obviously that's breaking things here.

Anyway the lesson learned here is that I should make PRs more frequently to avoid this in the future but lets chat later on the call about how we handle this particular PR.
